src/main/java/seedu/address/logic/parser/DeleteCommandParser.java

```diff
@@ -64,6 +68,13 @@ public DeleteCommand parse(String args) throws ParseException {
 
         if (argMultimap.prefixExist(PREFIX_MEDICAL)) {
             deletePersonDescriptor.setDeleteMedicalHistory();
+            if (argMultimap.getValue(PREFIX_MEDICAL).get().equals("")) {
+                deletePersonDescriptor.setMedicalHistory(new HashSet<>());
+            } else {
+                Set<MedicalHistory> medicalHistories = ParserUtil.parseMedicals(
+                        argMultimap.getAllValues(PREFIX_MEDICAL));
+                deletePersonDescriptor.setMedicalHistory(medicalHistories);
```

Comment:
`argMultimap` and `parseMedicals` already checks the empty field conditions.

Reply:
`argMultimap` creates a List of size 1 with an empty string, which will invoke the checks for valid input for medical history. Therefore, we require this `if else` statement.